crates/client/src/lib.rs

```diff
@@ -335,6 +337,21 @@ impl<R: RegistryStorage, C: ContentStorage> Client<R, C> {
             .await?;
 
         let mut heads = Vec::with_capacity(packages.len());
+        let operator_records = response.operator;
```

Comment:
So this looks relatively correct for validating operator records, there's a few things that need addressing here:

* The `FetchRequest` above is still passing a last operator record of `None`, which means the response will have _every_ record in the operator log that's in the given checkpoint, starting from the beginning of the log.

* This validation only works since we're assuming every record is being returned, which is why we start from a default `OperatorInfo`.

The way I would expect this to work is that, before fetching the logs, we load the operator info from storage (or create a default one). We pass the last known operator record identifier in `FetchRequest` (or `None` if the head is `None`). We then validate the records returned in the response.

Only after all validation passes should we persist the latest log states (for both packages and the operator log). That said, we should probably move the call to `prove_log_consistency` to above where we store the package logs and the code to store the operator log to below where we store the package logs.